File: problems/2379-minimum-recolors-to-get-k-consecutive-black-blocks/solution.java
class Solution {
    public int minimumRecolors(String blocks, int k) {
        int n = blocks.length();
        int minRecolors = Integer.MAX_VALUE;

        for (int i = 0; i <= n - k; i++) {
            int recolors = 0;
            for (int j = i; j < i + k; j++) {
                if (blocks.charAt(j) == 'W') {
                    recolors++;
                }
            }
            minRecolors = Math.min(minRecolors, recolors);
        }

        return minRecolors;
    }
}

File: problems/2379-minimum-recolors-to-get-k-consecutive-black-blocks/solution.c
int minimumRecolors(char* blocks, int k) {
    int n = 0;
    while (blocks[n] != '\0') {
        n++;
    }

    int min_recolors = k;
    int current_recolors = 0;

    for (int i = 0; i < k; i++) {
        if (blocks[i] == 'W') {
            current_recolors++;
        }
    }

    min_recolors = current_recolors;

    for (int i = k; i < n; i++) {
        if (blocks[i] == 'W') {
            current_recolors++;
        }
        if (blocks[i - k] == 'W') {
            current_recolors--;
        }
        if (current_recolors < min_recolors) {
            min_recolors = current_recolors;
        }
    }

    return min_recolors;
}